dwave/system/composites/cutoffcomposite.py

```diff
@@ -0,0 +1,329 @@
+
+        """
+        child = self.child
+        cutoff = self._cutoff
+        cutoff_vartype = self._cutoff_vartype
+        comp = self._comparison
+
+        if cutoff_vartype is dimod.SPIN:
+            original = poly.to_spin(copy=False)
+        else:
+            original = poly.to_binary(copy=False)
+
+        # remove all of the terms of order >= 2 that have a bias less than cutoff
+        new = type(poly)(((term, bias) for term, bias in original.items()
+                          if len(term) > 1 and not comp(abs(bias), cutoff)),
+                         cutoff_vartype)
+
+        # also include the linear biases for the variables in new
+        for v in new.variables:
+            term = v,
+            if term in original:
+                new[term] = original[term]
+
+        # everything else is isolated
+        isolated = list(original.variables.difference(new.variables))
+
+        if isolated and len(new) == 0:
+            # in this case all variables are isolated, so we just put one back
+            # to serve as the basis
+            term = isolated.pop(),
+            new[term] = original[term]
+
+        # get the samples from the child sampler and put them into the original vartype
+        sampleset = child.sample_poly(new, **kwargs).change_vartype(poly.vartype, inplace=True)
+
+        # we now need to add the isolated back in, in a way that minimizes
+        # the energy. There are lots of ways to do this but for now we'll just
+        # do one
+        if isolated:
+            samples, variables = _restore_isolated_higherorder(sampleset, poly, isolated)
+        else:
+            samples = sampleset.record.sample
+            variables = sampleset.variables
+
+        vectors = sampleset.data_vectors
+        vectors.pop('energy')  # we're going to recalculate the energy anyway
+
+        return dimod.SampleSet.from_samples_bqm((samples, variables), poly, **vectors)
+
+
+def _restore_isolated_higherorder(sampleset, poly, isolated):
+    """Return samples-like by adding isolated variables into sampleset in a
+    way that minimizes the energy (relative to the other non-isolated variables).
+
+    Isolated should be ordered.
+    """
+
+    low = -1 if poly.vartype is dimod.SPIN else 0
+
+    samples = sampleset.record.sample
+    variables = sampleset.variables
+
+    new_samples = np.empty((len(sampleset), len(isolated)), dtype=samples.dtype)
+
+    # we don't let the isolated variables interact with eachother for now because
+    # it will slow this down substantially
+    energies = {v: 0 for v in isolated}
+    for term, bias in poly.items():
+
+        isolated_components = term.intersection(isolated)
+
+        if not isolated_components:
+            continue
+
+        en = bias  # energy contribution of the term
+        for v in term:
+            if v in energies:
+                continue
+            en *= samples[:, sampleset.variables.index(v)]
+
+        for v in isolated_components:
```

Comment:
hghh, too slow if we put a for loop in for loop resolving as terms go, in case there are multiple isolated within a term?

Reply:
No, it would be approximately the same speed. I didn't do it for a couple reasons
1) This keeps things consistent with `CutOffComposite` where the affect would be noticeable
2) Right now the cutoff resolution is deterministic which is useful
3) If we're going to have allow effects between isolated variables then we also need a way of determining an order to resolve them

To that end, I think that would be in-scope for a set of more general post-processing tools.